src/projectile.cpp

```diff
@@ -0,0 +1,221 @@
+#include "projectile.h"
+
+#include "game.h"
+#include "map.h"
+#include "map_iterator.h"
+#include "explosion.h"
+#include "field.h"
+
+projectile::projectile() :
+    speed( 0 ), range( 0 ), drop( nullptr ), custom_explosion( nullptr )
+{ }
+
+projectile::~projectile()
+{
+}
+
+projectile::projectile( const projectile &other )
```

Comment:
The `operator=` operates on an `item&`, meaning it couldn't be pulled up to the header. Inlining a constructor that will then still expand into a function won't help with performance, especially if the compiler realizes the `operator=` is in the same compilation unit.